libs/api/user-profile/src/lib/user-profile.resolver.ts

```diff
@@ -0,0 +1,17 @@
+import { Args, Query, Resolver } from '@nestjs/graphql';
+import { UserProfileService } from './user-profile.service';
+import { UserProfileUniqueFields } from './types';
+import { UserProfileObjectType } from './user-profile.model';
+
+@Resolver()
+export class UserProfileResolver {
+  constructor(private readonly userProfileService: UserProfileService) {}
+
+  @Query(() => UserProfileObjectType)
+  async userProfile(
+    @Args('key', { type: () => String }) key: UserProfileUniqueFields,
+    @Args('value', { type: () => String }) value: string
```

Comment:
Maybe you can use a dedicated class if you have several arguments. See more [here](https://docs.nestjs.com/graphql/resolvers#dedicated-arguments-class).